doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,673 @@
+
+#### Scope
+The purpose of the test is testing of "ingress_discard_all" counter triggering on SONIC system, making sure that specific traffic drops correctly, according to sent packet and configured packet discards.
+Supported topologies:
+```
+t0
+t1
+t1-lag
+ptf32
+```
+
+#### Related DUT CLI commands
+| **Command**                                                      | **Comment** |
+|------------------------------------------------------------------|-------------|
+| show interfaces counters              | Check ```RX_DRP```                     |
+| counterpoll rif enable                | Enable RIF counters                    |
+| show interface counters rif           | Show RIF counters                      |
+| aclshow -a                            | Show ALC counters                      |
+
+#### General test flow
+Each test case will use the following port types:
+- VLAN (T0)
+- LAG (T0, T1-LAG)
+- Router (T1, T1-LAG, PTF32)
+
+##### Sent packet number:
+N - 5
+
+##### Repeat test scenario for all available port types depends on run topology (VLAN, LAG, Router)
+
+- Select two pairs of PTF and DUT ports considering topology: PTF_PORT[1] <--->DUT_PORT[1], PTF_PORT[2] <--->DUT_PORT[2]
+- Clear counters. Use CLI command "sonic-clear counters"
+- Inject N packet into PTF_PORT[1] (N - depends on test case)
+- Check "ingress_discard_all" counter incremented on N
+	- If counter was not incremented on N, test fails with expected message
+- Check other counters were incremented on N based on sent packet type, sent port and expected drop reason (depends on test case)
+	- If counter was not incremented on N, test fails with expected message
+- Check the packet was dropped by sniffing packet absence on PTF_PORT[2]
+
+
+#### Run test
+```
+py.test --inventory ../ansible/inventory --host-pattern [DEVICE] --module-path ../ansible/library/ --testbed [DEVICE-TOPO] --testbed_file ../ansible/testbed.csv --junit-xml=./target/test-reports/ --show-capture=no --log-cli-level debug -ra -vvvvv ingress_discard/test_ingress_discard.py
+```
+
+#### Test cases
+Each test case will be additionally validated by the loganalyzer utility.
+
+Each test case will run specific traffic to trigger specific discard reasone.
+
+Pytest fixture - "ptfadapter" is used to construct and send packets.
+
+After packet is sent using source port index, test framework waits during 5 seconds for specific packet did not appear, due to ingress packet drop, on one of the destination port indices.
+
+#### Test case #1
+Test objective
+
+Verify packet drops when SMAC and DMAC are equal.
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = 00:00:00:00:01:02
+  src = 00:00:00:00:01:02
+  type = 0x800
+...
+```
+
+Test steps
+- PTF host will send IP packet specifying identical src and dst MAC.
+- When packet reaches SONIC DUT, it should be dropped by "SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC"
```

Comment:
Why ```SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC```  SAI attribute is mentioned here? We don't have it yet.

Reply:
Same question for all bellow test cases, there are mentions of SAI attributes which we don't have yet.

Reply:
Fixed.